internal/backend/sessions/sessioncalls/calls.go

```diff
@@ -61,174 +59,186 @@ type calls struct {
 	return &calls{
-		z:                    z,
+		sl:                   sl,
 		config:               config,
 		svcs:                 svcs,
 		executorsForSessions: make(map[sdktypes.SessionID]*sdkexecutor.Executors, 16),
 	}
 }
 
 func (cs *calls) StartWorkers(ctx context.Context) error {
-	opts := cs.config.Temporal.Worker
-	opts.DisableRegistrationAliasing = true
-	opts.OnFatalError = func(err error) { cs.z.Error("temporal worker error", zap.Error(err)) }
-	opts.DisableWorkflowWorker = true // these workers serve only activities.
-
-	cs.generalWorker = worker.New(cs.svcs.TemporalClient(), taskQueueName, opts)
-	cs.uniqueWorker = worker.New(cs.svcs.TemporalClient(), uniqueWorkerCallTaskQueueName(), opts)
-
-	cs.generalWorker.RegisterActivityWithOptions(
-		cs.sessionCallActivity,
-		activity.RegisterOptions{
-			Name:                          callActivityName,
-			DisableAlreadyRegisteredCheck: true,
-		},
-	)
-
-	cs.uniqueWorker.RegisterActivityWithOptions(
-		cs.sessionCallActivity,
-		activity.RegisterOptions{
-			Name:                          callActivityName,
-			DisableAlreadyRegisteredCheck: true,
-		},
-	)
-
-	if err := cs.generalWorker.Start(); err != nil {
-		return err
+	l := cs.sl.Desugar()
+	cs.generalWorker = temporalclient.NewWorker(l.Named("sessionscallsworker"), cs.svcs.Temporal(), generalTaskQueueName, cs.config.GeneralWorker)
+	cs.uniqueWorker = temporalclient.NewWorker(l.Named("sessionscallsworker"), cs.svcs.Temporal(), uniqueWorkerCallTaskQueueName(), cs.config.UniqueWorker)
+
+	cs.registerActivities()
+
+	if w := cs.generalWorker; w != nil {
+		if err := w.Start(); err != nil {
+			return err
+		}
+	}
+
+	if w := cs.uniqueWorker; w != nil {
+		if err := w.Start(); err != nil {
+			return err
+		}
 	}
 
-	return cs.uniqueWorker.Start()
+	return nil
 }
 
-func (cs *calls) Call(ctx workflow.Context, params *CallParams) (sdktypes.SessionCallAttemptResult, error) {
+func (cs *calls) Call(wctx workflow.Context, params *CallParams) (sdktypes.SessionCallAttemptResult, error) {
 	spec := params.CallSpec
+	sid := params.SessionID
 
 	fnv, _, _ := spec.Data()
 	fnvf := fnv.GetFunction()
 
 	seq := spec.Seq()
 
-	z := cs.z.With(zap.String("session_id", params.SessionID.String()), zap.Uint32("seq", seq), zap.Any("v", fnv))
+	sl := cs.sl.With("session_id", sid, "seq", seq, "v", fnv)
+
+	wctx = workflow.WithActivityOptions(wctx, cs.config.activityConfig().ToOptions(generalTaskQueueName))
 
-	// TODO: If replaying, make sure arguments are the same?
-	if err := workflow.ExecuteLocalActivity(
-		ctx, cs.svcs.DB.CreateSessionCall, params.SessionID, spec,
-	).Get(ctx, nil); err != nil {
-		return sdktypes.InvalidSessionCallAttemptResult, fmt.Errorf("db.create_call: %w", err)
+	if err := workflow.ExecuteActivity(
+		wctx,
+		createSessionCallActivityName,
+		sid,
+		spec,
+	).Get(wctx, nil); err != nil {
+		return sdktypes.InvalidSessionCallAttemptResult, fmt.Errorf("create_call: %w", err)
```

Comment:
`if` with a multi-line condition is not readable. This is much better. Same comment for the `if` below.

```suggestion
	future := workflow.ExecuteActivity(wctx, createSessionCallActivityName, sid, spec)
	if future.Get(wctx, nil); err != nil {
		return sdktypes.InvalidSessionCallAttemptResult, fmt.Errorf("create_call: %w", err)
```

Reply:
done